src/main/java/thePackmaster/powers/summonerspellspack/TeleportBotPower.java

```diff
@@ -0,0 +1,47 @@
+package thePackmaster.powers.summonerspellspack;
+
+import com.megacrit.cardcrawl.actions.utility.DiscardToHandAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.core.AbstractCreature;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import thePackmaster.powers.AbstractPackmasterPower;
+
+import static thePackmaster.SpireAnniversary5Mod.makeID;
+import static thePackmaster.util.Wiz.p;
+
+public class TeleportBotPower extends AbstractPackmasterPower {
+    public static final String POWER_ID = makeID("TeleportBotPower");
+    public static final String NAME = CardCrawlGame.languagePack.getPowerStrings(POWER_ID).NAME;
+    public static final String[] DESCRIPTIONS = CardCrawlGame.languagePack.getPowerStrings(POWER_ID).DESCRIPTIONS;
+
+    public TeleportBotPower(AbstractCreature owner, int amount) {
+        super(POWER_ID, NAME, PowerType.BUFF, false, owner, amount);
+    }
+
+    @Override
+    public void atStartOfTurnPostDraw() {
+        boolean flashed = false;
+        for (int i = 0; i < this.amount; i++) {
+            if (!p().discardPile.isEmpty()) {
+                AbstractCard c = p().discardPile.group.get(0);
+                addToBot(new DiscardToHandAction(c));
```

Comment:
The action needs to finish before the card is actually in hand, so this only ever works on one card, right?

Reply:
I tested and you're right. I've rewritten this code to no longer pull the 0th card but the ith card, if i is smaller than the discard pile size.